builtin/credential/okta/path_config.go

```diff
@@ -290,36 +293,128 @@ func (b *backend) pathConfigExistenceCheck(ctx context.Context, req *logical.Req
+	Client() (*oktanew.APIClient, context.Context)
 	NewRequest(method string, url string, body interface{}) (*http.Request, error)
 	Do(req *http.Request, v interface{}) (interface{}, error)
 }
 
 type oktaShimNew struct {
-	client *oktanew.Client
+	cfg    *oktanew.Configuration
+	client *oktanew.APIClient
 	ctx    context.Context
 }
 
-func (new *oktaShimNew) Client() (*oktanew.Client, context.Context) {
+func (new *oktaShimNew) Client() (*oktanew.APIClient, context.Context) {
 	return new.client, new.ctx
 }
 
 func (new *oktaShimNew) NewRequest(method string, url string, body interface{}) (*http.Request, error) {
 	if !strings.HasPrefix(url, "/") {
 		url = "/api/v1/" + url
 	}
-	return new.client.GetRequestExecutor().NewRequest(method, url, body)
+
+	// reimplementation of RequestExecutor.NewRequest() in v2 of okta-golang-sdk
+	var buff io.ReadWriter
+	if body != nil {
+		switch v := body.(type) {
+		case []byte:
+			buff = bytes.NewBuffer(v)
+		case *bytes.Buffer:
+			buff = v
+		default:
+			buff = &bytes.Buffer{}
+			// need to create an encoder specifically to disable html escaping
+			encoder := json.NewEncoder(buff)
+			encoder.SetEscapeHTML(false)
+			err := encoder.Encode(body)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	url = new.cfg.Okta.Client.OrgUrl + url
+	req, err := http.NewRequest(method, url, buff)
+	if err != nil {
+		return nil, err
+	}
+	//
+	var auth oktanew.Authorization
+	//
+	switch new.cfg.Okta.Client.AuthorizationMode {
+	case "SSWS":
+		auth = oktanew.NewSSWSAuth(new.cfg.Okta.Client.Token, req)
+	case "Bearer":
+		auth = oktanew.NewBearerAuth(new.cfg.Okta.Client.Token, req)
+	case "PrivateKey":
+		auth = oktanew.NewPrivateKeyAuth(oktanew.PrivateKeyAuthConfig{
+			// TokenCache:       new.cfg., hmm
+			HttpClient:       new.cfg.HTTPClient,
+			PrivateKeySigner: new.cfg.PrivateKeySigner,
+			PrivateKey:       new.cfg.Okta.Client.PrivateKey,
+			PrivateKeyId:     new.cfg.Okta.Client.PrivateKeyId,
+			ClientId:         new.cfg.Okta.Client.ClientId,
+			OrgURL:           new.cfg.Okta.Client.OrgUrl,
+			Scopes:           new.cfg.Okta.Client.Scopes,
+			MaxRetries:       new.cfg.Okta.Client.RateLimit.MaxRetries,
+			MaxBackoff:       new.cfg.Okta.Client.RateLimit.MaxBackoff,
+			Req:              req,
+		})
+	case "JWT":
+		auth = oktanew.NewJWTAuth(oktanew.JWTAuthConfig{
+			HttpClient:      new.cfg.HTTPClient,
+			OrgURL:          new.cfg.Okta.Client.OrgUrl,
+			Scopes:          new.cfg.Okta.Client.Scopes,
+			ClientAssertion: new.cfg.Okta.Client.ClientAssertion,
+			MaxRetries:      new.cfg.Okta.Client.RateLimit.MaxRetries,
+			MaxBackoff:      new.cfg.Okta.Client.RateLimit.MaxBackoff,
+			Req:             req,
+		})
+	default:
+		return nil, fmt.Errorf("unknown authorization mode %v", new.cfg.Okta.Client.AuthorizationMode)
+	}
+
+	err = auth.Authorize("POST", url)
+	if err != nil {
+		return nil, err
+	}
+
+	// req.Header.Add("User-Agent", NewUserAgent(re.config).String())
```

Comment:
Could we remove or have a bit more details to why it's left out/commented?

Reply:
removed